diff -ur linux-2.6.31.6_orig/fs/compat_ioctl.c linux-2.6.31.6/fs/compat_ioctl.c
--- linux-2.6.31.6_orig/fs/compat_ioctl.c	2009-11-10 01:32:31.000000000 +0100
+++ linux-2.6.31.6/fs/compat_ioctl.c	2009-11-21 12:35:28.000000000 +0100
@@ -99,6 +99,7 @@
 
 #include <linux/usb.h>
 #include <linux/usbdevice_fs.h>
+#include <linux/vhci-hcd.h>
 #include <linux/nbd.h>
 #include <linux/random.h>
 #include <linux/filter.h>
@@ -2423,6 +2424,12 @@
 COMPATIBLE_IOCTL(USBDEVFS_REAPURB32)
 COMPATIBLE_IOCTL(USBDEVFS_REAPURBNDELAY32)
 COMPATIBLE_IOCTL(USBDEVFS_CLEAR_HALT)
+/* USB Virtual Host Controller Interface */
+COMPATIBLE_IOCTL(VHCI_HCD_IOCREGISTER)
+COMPATIBLE_IOCTL(VHCI_HCD_IOCPORTSTAT)
+COMPATIBLE_IOCTL(VHCI_HCD_IOCFETCHWORK)
+COMPATIBLE_IOCTL(VHCI_HCD_IOCGIVEBACK32)
+COMPATIBLE_IOCTL(VHCI_HCD_IOCFETCHDATA32)
 /* NBD */
 ULONG_IOCTL(NBD_SET_SOCK)
 ULONG_IOCTL(NBD_SET_BLKSIZE)
